modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/deployer/ObjectDefinitionDeployerImpl.java

```diff
@@ -398,16 +398,7 @@ public List<ServiceRegistration<?>> deploy(
 					"com.liferay.portlet.company",
 					objectDefinition.getCompanyId()
 				).put(
-					"com.liferay.portlet.display-category",
-					() -> {
-						if (objectDefinition.isPortlet() &&
-							!objectDefinition.isRootDescendantNode()) {
-
-							return "category.object";
-						}
-
-						return "category.hidden";
-					}
```

Comment:
Hey @leilanyaragao 

We can't remove this conditional. This conditional was created in this [story](https://liferay.atlassian.net/browse/LPS-143122). So, when he definition isPortlet, we add the definition on object category. Otherwise, we hidden this definition. Futhermore, when the definition belongs a root concept, this definition can't portlet as well.

So, I found this [spike](https://liferay.atlassian.net/browse/LPS-188440?focusedCommentId=2185786) of @devpedrotavares with 3 approach to solve this issue. Could you take a look, please?